examples/import_yara_ruleset.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/local/bin/python
+# -*- coding: utf-8 -*-
+# Copyright © 2020 The vt-py authors. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Import YARA rulesets to a VirusTotal account.
+
+This script imports either a ruleset in a file or all ruleset files in a given
+directory. These imported YARA rules can be used in VT Hunting.
+
+Read more:
+https://www.virustotal.com/gui/hunting-overview
+https://developers.virustotal.com/v3.0/reference#livehunt
+https://support.virustotal.com/hc/en-us/articles/360000363717-VT-Hunting
+"""
+
+import argparse
+import asyncio
+import os
+import vt
+import sys
+
+
+async def get_rules_files(queue, path):
+  """Finds which rules will be uploaded to VirusTotal."""
+  if os.path.isfile(path):
+    await queue.put(path)
+    return
+
+  with os.scandir(path) as it:
+    for entry in it:
+        if not entry.name.startswith('.') and entry.is_file():
+            await queue.put(entry.path)
+
+
+async def upload_rules(queue, apikey, enable):
+  """Uploads selected files to VirusTotal."""
+  async with vt.Client(apikey) as client:
+    while not queue.empty():
+      file_path = await queue.get()
+      with open(file_path) as f:
+        ruleset = vt.Object(
+            obj_type='hunting_ruleset',
+            obj_attributes={
+                'name': os.path.basename(file_path),
+                'enabled': enable,
+                'rules': f.read()})
+
+        try:
+          await client.post_object_async(
+              path='/intelligence/hunting_rulesets', obj=ruleset)
+          print(f'File {file_path} uploaded.')
+        except vt.error.APIError as e:
+          print(f'Error uploading {file_path}: {e}')
```

Comment:
This piece of code doesn't need to be inside the above with statement

Reply:
Fixed in commit 1e3a6e3